File: src/tetris/OPiece.java
package tetris;

import javalib.worldimages.Posn;

class OPiece extends APiece {
	// root posn top left
	static boolean[][] OPIECE_UP = new boolean[][] {
			{ true, true, false, false },
			{ true, true, false, false },
			{ false, false, false, false },
			{ false, false, false, false }
	};
	
	static boolean[][] OPIECE_COPY = OPIECE_UP.clone();
	static CycleIndex OPIECE_ROTATIONS = new CycleIndex(OPIECE_COPY, OPIECE_COPY, OPIECE_COPY, OPIECE_COPY);
	
	
	OPiece(Posn posn) {
		super(posn, OPIECE_ROTATIONS, Tetrimino.O);
	}

	public boolean hasSpun(Board b) {
		return false;
	}
	
	public boolean[][] rotInitialState(String s) {
		return OPIECE_UP;
	}


	
}

File: typescript/src/pieces/OPiece.ts
import { Posn } from 'impworld';
import { APiece } from './APiece';
import { CycleIndex } from './CycleIndex';
import { Tetrimino } from './Tetrimino';
import { Board } from '@/board';

const OPIECE_UP: boolean[][] = [
  [true, true, false, false],
  [true, true, false, false],
  [false, false, false, false],
  [false, false, false, false],
];

const OPIECE_COPY: boolean[][] = [...OPIECE_UP];

const OPIECE_ROTATIONS: CycleIndex = new CycleIndex(
  OPIECE_COPY,
  OPIECE_COPY,
  OPIECE_COPY,
  OPIECE_COPY,
);

export class OPiece extends APiece {
  constructor(posn: Posn) {
    super(posn, OPIECE_ROTATIONS, Tetrimino.O);
  }

  public hasSpun(b: Board): boolean {
    return false;
  }

  public rotInitialState(s: string): boolean[][] {
    return OPIECE_UP;
  }
}
